packages/core/parcel-bundler/src/Pipeline.js

```diff
@@ -17,16 +17,30 @@ class Pipeline {
     }
 
     let asset = this.parser.getAsset(path, options);
-    let generated = await this.processAsset(asset);
+    let error = null;
     let generatedMap = {};
-    for (let rendition of generated) {
-      generatedMap[rendition.type] = rendition.value;
+    try {
+      let generated = await this.processAsset(asset);
+      for (let rendition of generated) {
+        generatedMap[rendition.type] = rendition.value;
+      }
+    } catch (err) {
+      // Error instances need to be converted to plain JS objects.
```

Comment:
I _think_ this is because in some flows, the value returned from this function passes through `JSON.Stringify` which results in an empty object. I couldn't track that code down, however, I just observed the behavior, so I'm quite curious about what I missed. Maybe something related to caching?

Reply:
I also considered handling this transformation in `Asset.generateErrorMessage`, but I was worried someone might override it to return an `Error` type and have difficulty figuring out why errors were behaving oddly.